crates/joystick/src/common.rs

```diff
@@ -0,0 +1,75 @@
+use evdev::KeyCode;
+use std::time::Duration;
+
+pub static TAP_DURATION: u64 = 50;
+
+pub enum KeyAction {
+    Press,
+    Release,
+}
+
+pub trait JoystickInterface {
+    fn press_a(&mut self);
+    fn press_b(&mut self);
+    fn press_x(&mut self);
+    fn press_y(&mut self);
+    fn press_lt(&mut self);
+    fn press_rt(&mut self);
+    fn press_lt2(&mut self);
+    fn press_rt2(&mut self);
+    fn press_select(&mut self);
+    fn press_start(&mut self);
+    fn press_dpad_up(&mut self);
+    fn press_dpad_down(&mut self);
+    fn press_dpad_left(&mut self);
+    fn press_dpad_right(&mut self);
+
+    fn release_a(&mut self);
+    fn release_b(&mut self);
+    fn release_x(&mut self);
+    fn release_y(&mut self);
+    fn release_lt(&mut self);
+    fn release_rt(&mut self);
+    fn release_lt2(&mut self);
+    fn release_rt2(&mut self);
+    fn release_select(&mut self);
+    fn release_start(&mut self);
+    fn release_dpad_up(&mut self);
+    fn release_dpad_down(&mut self);
+    fn release_dpad_left(&mut self);
+    fn release_dpad_right(&mut self);
+
+    fn tap_a(&mut self);
```

Comment:
All the tap functions could be implemented here on the trait, since they are all basically just:

1. press (x)
2. wait some duration
3. release (x)
4. Optionally, wait some duration (when tapping)